File: olymp/vkcup_4/d.py
prior = [3,3,4,4,3,3]

def get_prior(x, y):
    return prior[y] - x // 2

px, py, pa = 0,0,0
m = []

for x in range(6):
    s = input()
    m.append(list(s))
    s = s.replace('-', '')
    for y in range(6):
        p = get_prior(x, y)
        if s[y] == '.' and p > pa:
            px, py, pa = x, y, p

if py > 3:
    py += 2
elif py > 1:
    py +=1

m[px][py] = 'P'

for i in m:
    print(''.join(i))

File: olymp/vkcup_4/d.rb
prior = [ 3, 3, 4, 4, 3, 3 ]

px = 0
py = 0
pa = 0

m = []

j = 0
while j < 6
	s = gets.chomp
    m = m + [s]
	for i in (0..8)
		if s[i] == '.'
			y = i
			if y > 4
				y = y - 2
			else
				if y > 1
					y = y -1
				end
			end
			
			a = prior[y] - j / 2
			if a > pa
				px = j
				py = i
				pa = a
			end
        end
	end
    j = j + 1
end

m[px][py] = "P"

for i in (0..5)
    print m[i], "\n"
end


